dkg/pedersen/handler.go

```diff
@@ -119,64 +110,83 @@ func (h *Handler) Stream(out mino.Sender, in mino.Receiver) error {
 	// messages to the other nodes, and then we might get their messages before
 	// the start message.
 
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
+	// We make sure not additional request is accepted if a setup is in
+	// progress.
+	h.Lock()
+	if !h.startRes.Done() && h.running {
```

Comment:
```suggestion
	if h.running {
```
wouldn't this be sufficient / cleaner ?
and if not, documenting why might be useful to future programmers :-)

Reply:
yes, it works too